Sources/PostgresNIO/New/Connection State Machine/PreparedStatementStateMachine.swift

```diff
@@ -0,0 +1,68 @@
+import NIOCore
+
+struct PreparedStatementStateMachine {
+    enum State {
+        case preparing([PreparedStatementContext])
```

Comment:
why is this an array?

Reply:
So that we can send the `PREPARE` command only once if there are are multiple attempts to execute the same prepared statement before the command reaches the server.

I thought we said this is something that can happen and we want to handle